### PR TITLE
Adds a noAutoPlayVideo configuration option (used in testing).

### PR DESCRIPTION
**Pull request by gpolitis**

### CODE REVIEW

**Comment by gpolitis** [comment]
This is going to be used mostly with maleus (our selenium-based load testing tool for testing the new bridge), so I focused on implementing the option for web only.

**Comment by gpolitis** [comment]
I was talking to @damencho and he's fine with the changes. @saghul, @virtuacoplenny or @hristoterezov let me know if you need me to change anything.

File: react/features/base/media/components/web/Video.js [hunk 1]
```diff
@@ -131,7 +133,7 @@ class Video extends Component<Props> {
     render() {
         return (
             <video
-                autoPlay = { true }
+                autoPlay = { !config.testing.noAutoPlayVideo }
```
Comment (hristoterezov) [inline react/features/base/media/components/web/Video.js]: I think for the reactified files we tend to get config from redux.

File: react/features/base/media/components/web/VideoTrack.js [hunk 2]
```diff
@@ -53,7 +55,7 @@ class VideoTrack extends AbstractVideoTrack<Props> {
     render() {
         return (
             <Video
-                autoPlay = { true }
+                autoPlay = { !config.testing.noAutoPlayVideo }
```
Comment (hristoterezov) [inline react/features/base/media/components/web/VideoTrack.js]: I think for the reactified files we tend to get config from redux.

File: react/features/large-video/components/LargeVideo.web.js [hunk 3]
```diff
@@ -51,7 +52,7 @@ export default class LargeVideo extends Component<{}> {
                       */}
                     <div id = 'largeVideoWrapper'>
                         <video
-                            autoPlay = { true }
+                            autoPlay = { !config.testing.noAutoPlayVideo }
```
Comment (hristoterezov) [inline react/features/large-video/components/LargeVideo.web.js]: I think for the reactified files we tend to get config from redux

File: react/features/base/media/components/web/Video.js [hunk 4]
```diff
@@ -200,4 +206,22 @@ class Video extends Component<Props> {
...
+ * @private
+ * @returns {{
+ *     noAutoPlayVideo: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const testingConfig = state['features/base/config'];
+
+    return {
+        noAutoPlayVideo: testingConfig && testingConfig.noAutoPlayVideo
```
Comment (hristoterezov) [inline react/features/base/media/components/web/Video.js]: Isn't this going to be actually `config.noAutoPlayVideo` and not `config.testing.noAutoPlayVideo`?

File: react/features/base/media/components/web/VideoTrack.js [hunk 5]
```diff
@@ -64,4 +69,22 @@ class VideoTrack extends AbstractVideoTrack<Props> {
...
+ * @private
+ * @returns {{
+ *     noAutoPlayVideo: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const testingConfig = state['features/base/config'];
+
+    return {
+        noAutoPlayVideo: testingConfig && testingConfig.noAutoPlayVideo
```
Comment (hristoterezov) [inline react/features/base/media/components/web/VideoTrack.js]: Isn't this going to be actually config.noAutoPlayVideo and not config.testing.noAutoPlayVideo?

File: react/features/large-video/components/LargeVideo.web.js [hunk 6]
```diff
@@ -63,3 +72,24 @@ export default class LargeVideo extends Component<{}> {
...
+ * @private
+ * @returns {{
+ *     noAutoPlayVideo: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const testingConfig = state['features/base/config'];
+
+    return {
+        noAutoPlayVideo: testingConfig && testingConfig.noAutoPlayVideo
```
Comment (hristoterezov) [inline react/features/large-video/components/LargeVideo.web.js]: Isn't this going to be actually config.noAutoPlayVideo and not config.testing.noAutoPlayVideo?

**Comment by gpolitis** [comment]
Thanks @hristoterezov, I've updated the code. Could you please take another look?

File: react/features/base/media/components/web/Video.js (hunk 4 above)
Comment (gpolitis) [reply react/features/base/media/components/web/Video.js]: You're absolutely right, I forgot to updated my deployment for testing after I modified the code with your previous recommendation.

File: react/features/base/media/components/web/Video.js [hunk 7]
```diff
@@ -200,4 +207,22 @@ class Video extends Component<Props> {
...
+ * @private
+ * @returns {{
+ *     _noAutoPlayVideo: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const testingConfig = state['features/base/config'].testing;
+
+    return {
+        _noAutoPlayVideo: testingConfig && testingConfig.noAutoPlayVideo
```
Comment (saghul) [inline react/features/base/media/components/web/Video.js]: You can use nullish coalescing here: `testingConfig?.noAutoPlayVideo`

File: react/features/base/media/components/web/VideoTrack.js [hunk 8]
```diff
@@ -64,4 +71,22 @@ class VideoTrack extends AbstractVideoTrack<Props> {
...
+ * @private
+ * @returns {{
+ *     _noAutoPlayVideo: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const testingConfig = state['features/base/config'].testing;
+
+    return {
+        _noAutoPlayVideo: testingConfig && testingConfig.noAutoPlayVideo
```
Comment (saghul) [inline react/features/base/media/components/web/VideoTrack.js]: ditto

File: react/features/large-video/components/LargeVideo.web.js [hunk 9]
```diff
@@ -63,3 +73,24 @@ export default class LargeVideo extends Component<{}> {
...
+ * @private
+ * @returns {{
+ *     _noAutoPlayVideo: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const testingConfig = state['features/base/config'].testing;
+
+    return {
+        _noAutoPlayVideo: testingConfig && testingConfig.noAutoPlayVideo
```
Comment (saghul) [inline react/features/large-video/components/LargeVideo.web.js]: ditto

File: react/features/base/media/components/web/Video.js (hunk 7 above)
Comment (gpolitis) [reply react/features/base/media/components/web/Video.js]: me likey, will do, thanks @saghul

**Comment by gpolitis** [comment]
Please don't merge this yet, it breaks the device selection dialog (as the VideoInputPreview component inherits from the Video component) and haven't figured out how to fix it yet.

**Comment by gpolitis** [comment]
The problem is that the device dialog popup makes use of the VideoInputPreview component <- extends the Video component <- that uses redux BUT we don't have redux in the dialog and it's not clear that we want to add it there. So I removed redux from the Video component: the device dialog simply won't respect the `_noAutoPlayVideo` configuration setting, but that's okay.

**Pull request merged**